_posts/2024-12-01-mesh3-improvements.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title: "Tetrahedral Mesh Generation improvements"
+description: "Tetrahedral Mesh Generation improvements in CGAL-6.0.1"
+category:
+tags: [""]
+---
+{% include JB/setup %}
+
+<h3><a href="https://geometryfactory.com/who-we-are/">Jane Tournois</a></h3>
+<h4><a href="http://www.geometryfactory.com" target="_blank">GeometryFactory</a></h4>
+
+
+<p>Introduced in CGAL 3.5, the package
+<a href="https://doc.cgal.org/latest/Manual/packages.html#PkgMesh3" text="3D Mesh Generation">3D Mesh Generation</a>
+is designed to generate a isotropic simplicial meshes discretizing 3D domains.
+Since then, multiple improvements in mesh quality, performances, and versatility have been made available in CGAL.
+Lately, CGAL authors have focused more on the quality of the output mesh, the ability of the
+mesh generation algorithm to represent the input surface in a more accurate way, for a wider range of input data types.
+</p>
+
+<br>
+<p>Recent releases of CGAL have introduced improvements to the 3D Tetrahedral Mesh Generation package:
+detection and protection of triple lines from labeled images,
+generation and use of weighted labeled images,
+and a new approximation error criterion for polyline features.</p>
+
+
+<br>
+<h3>Triple lines from labeled images</h3>
+
+<p>CGAL's tetrahedral mesh generation algorithm is able to compute tetrahedral meshes from 3D segmented images.
+For all types of inputs, detection and construction of one-dimensional features must be done a priori for the
+feature protection algorithm to be able to represent them properly in the generated mesh.
+In 3D segmented images, such features correspond to the intersections of the bounding box of the image with the surfaces
+defined by the image, as well as polylines that lie at the intersection of three or more subdomains (including the outside).</p>
+
+<p>The article
+<a href="https://opus4.kobv.de/opus4-zib/frontdoor/index/index/docId/274"><i>Hege, H. C., Seebass, M., Stalling, D., & Zöckler, M. (1997),
+A generalized marching cubes algorithm based on non-binary classifications</i></a>, provides a list of possible voxel configurations.
+This description has been generalized for tetrahedral mesh generation needs, and
+implemented in CGAL.</p>
+
+<p>After feature lines detection, similarly to what is done for polyhedral surfaces, the polyline features
+are added to the feature graph of the domain for feature protection.</p>
+
+<p>Two detection functors are provided :
+<ul>
+  <li><i>CGAL::Detect_features_in_image()</i> detects and constructs all
```

Comment:
In other posts I saw <code>..</code> being used.

Reply:
like `CGAL::Detect_features_in_image(..)` ?

Reply:
`<code>...</code>`